packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -887,5 +914,77 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
         });
       });
     });
+
+    describe('@hbarlimiter-batch3 BASIC Tier', () => {
+      const accounts: AliasAccount[] = [];
+
+      beforeEach(async function () {
+        logger.info(`${requestDetails.formattedRequestId} Creating accounts`);
+        logger.info(
+          `${requestDetails.formattedRequestId} HBAR_RATE_LIMIT_TINYBAR: ${ConfigService.get(
+            'HBAR_RATE_LIMIT_TINYBAR',
+          )}`,
+        );
+
+        const initialAccount: AliasAccount = global.accounts[0];
+
+        const neededAccounts: number = 3;
+        accounts.push(
+          ...(await Utils.createMultipleAliasAccounts(
+            mirrorNode,
+            initialAccount,
+            neededAccounts,
+            initialBalance,
+            requestDetails,
+          )),
+        );
+        global.accounts.push(...accounts);
+        const basicPlans = await hbarSpendingPlanRepository.findAllActiveBySubscriptionTier(
+          [SubscriptionTier.BASIC],
+          requestDetails,
+        );
+        for (const plan of basicPlans) {
+          await hbarSpendingPlanRepository.delete(plan.id, requestDetails);
+          await ethAddressSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
+          await ipSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
+        }
+      });
+
+      it('should eventually exhaust the hbar limit for a BASIC user after multiple deployments of large contracts, and not throw an error', async function () {
+        let expectedTxCost = 0;
+        let deploymentCounts = 0;
+        let hbarSpendingPlan: IDetailedHbarSpendingPlan | null = null;
+
+        for (deploymentCounts = 0; deploymentCounts < 3; deploymentCounts++) {
+          const tx = await deployContract(largeContractJson, accounts[2].wallet);
+          await tx.waitForDeployment();
+
+          expectedTxCost ||= await getExpectedCostOfLastLargeTx(largeContractJson.bytecode);
+
+          if (!hbarSpendingPlan) {
+            const ethSpendingPlan = await ethAddressSpendingPlanRepository.findByAddress(
+              accounts[2].wallet.address,
+              requestDetails,
+            );
+            hbarSpendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
+              ethSpendingPlan.planId,
+              requestDetails,
+            );
+          }
+
+          await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts + 1, expectedTxCost);
+        }
+        // Verify that the amount spent exceeds the HBAR limit
+        const amountSpent = await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts, expectedTxCost);
```

Comment:
after the loop hbarSpendingPlan can still be null in cases that no loop has got through. Please add this block above this line so the null error can be discarded
```typescript
      if (!hbarSpendingPlan) {
             const ethSpendingPlan = await ethAddressSpendingPlanRepository.findByAddress(
             accounts[2].wallet.address,
             requestDetails,
             );
             hbarSpendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
                 ethSpendingPlan.planId,
                 requestDetails,
        );
     }
```

Reply:
Done.